### PR TITLE
11.0 imp alipay nla rde

### PR DESCRIPTION
**Pull request by rdeodoo**
task-40221

### CODE REVIEW

File: addons/payment_transfer/models/payment.py [hunk 1]
```diff
@@ -53,6 +53,11 @@ def write(self, values):
             values['post_msg'] = self._format_transfer_data()
         return super(TransferPaymentAcquirer, self).write(values)
 
+    def _get_feature_support(self):
```
[the remark this answers is not in the document]
Comment (tbe-odoo) [reply addons/payment_transfer/models/payment.py]: Yup, you are right but this is what we want.
Custom with transfer is just a "duplicate" of "Wire Transfer".

The customer has to change the provider of Custom to access the payment acquirer features.

**Comment by rdeodoo** [comment]
@tbe-odoo Last 2 commits of the PR fixed what we discussed (Block payment when express_checkout and currency not CNY + Repair error handler in payment), may I have your blessing? :pray:

And about some _logger getting translated, I guess it is a problem since we won't be able to retrieve the log in the code once translated?
eg:
```
if sign != sign_check:
    error_msg = _('Alipay: invalid sign, received %s, computed %s, for data %s') % (sign, sign_check, data)
    _logger.info(error_msg)
    raise ValidationError(error_msg)
```

I get that I is convenient to avoid writing the same text twice but it should be like below or it doesn't matter?
```
if sign != sign_check:
     error_msg = 'Alipay: invalid sign, received %s, computed %s, for data %s' % (sign, sign_check, data)
    _logger.info(error_msg)
    raise ValidationError(_('Alipay: invalid sign, received %s, computed %s, for data %s') % (sign, sign_check, data))
```

Thanks

**Comment by rdeodoo** [comment]
Task https://www.odoo.com/web#id=40221&view_type=form&model=project.task&menu_id=

Everything seems to work correctly now:
- Return url and notify url works fine
- Express checkout and Standard checkout works fine. Note that we forbid express checkout with currency not CNY
- Payment with Alipay mobile apps with QR code works fine
- Tried all the flows with china and arabic language both in backend and front end and works fine
- Fees works fine. Note that this does not work like paypal which you can give the SO amount and the additionnal fee that will be displayed separately on paypal checkout. Fees will be listed on a separate line like 'Frais de dossier'. In alipay you just sum the fees and the amount in odoo and send the total to alipay that will just ask the customer for the amount without any indication that part of the amount is a fee.
- Since Alipay does not support s2s, the form field are correctly hidden. Note that others provider not s2s now also have the fields hidden.

**Comment by rdeodoo** [comment]
@tbe-odoo I fixed it plus the prevent double validation in last commit or this PR

**Comment by tbe-odoo** [comment]
Hello @tde-banana-odoo , as you are the master of payment acquirers, may I ask you to review this please :)

**Comment by rdeodoo** [comment]
@tbe-odoo Hi, how is this PR doing in China?
Any plan to merge this in standard?

**Comment by dbo-odoo** [comment]
Will be finished in #21855

**Pull request closed without merging**